src/Websites/Websites/Cmdlets/DeploymentSlots/SetAzureWebAppSlot.cs

```diff
@@ -260,6 +264,10 @@ public override void ExecuteCmdlet()
                         //Update WebApp object after site update
                         WebApp = new PSSite(WebsitesClient.GetWebApp(ResourceGroupName, Name, null));
                     }
+                    if (parameters.Contains("HostNames"))
+                    {
+                        WebsitesClient.AddCustomHostNames(ResourceGroupName, location, Name, HostNames);
```

Comment:
Can you check this - shouldn't you need to pass the slot name as well here? the Name is going to be just the site name.